Lib/fontTools/varLib/instancer.py

```diff
@@ -1284,6 +1298,158 @@ def instantiateVariableFont(
     return varfont
 
 
+def axisValuesFromAxisLimits(stat, axisLimits):
+
+    axisRecords = stat.table.DesignAxisRecord.Axis
+    axisValues = stat.table.AxisValueArray.AxisValue
+
+    format4 = [a for a in axisValues if a.Format == 4]
+    nonformat4  = [a for a in axisValues if a not in format4]
+    axisValues = format4 + nonformat4
+
+    axisOrder = {a.AxisOrdering: a.AxisTag for a in axisRecords}
+    pinnedAxes = set(k for k, v in axisLimits.items() if isinstance(v, (float, int)))
+
+    results, seen_axes = [], set()
+    for axisValue in axisValues:
+
+        # Ignore axisValue if it has ELIDABLE_AXIS_VALUE_NAME flag enabled.
+        # Enabling this flag will hide the axisValue in application font menus.
+        # TODO this is too greedy! we need to retain wght axisValues
+        if axisValue.Flags == 2:
+            continue
+
+        if axisValue.Format == 4:
+            axisIndexes = set(r.AxisIndex for r in axisValue.AxisValueRecord)
+            if seen_axes - axisIndexes != seen_axes:
+                continue
+            # TODO fix dup appends 
+            for rec in axisValue.AxisValueRecord:
+                axisTag = axisOrder[rec.AxisIndex]
+                if axisTag not in pinnedAxes:
+                    continue
+                if rec.Value == axisLimits[axisTag]:
+                    seen_axes.add(rec.AxisIndex)
+                    results.append((rec.AxisIndex, axisValue))
+
+        elif axisValue.Format in (1, 3):
+            axisTag = axisOrder[axisValue.AxisIndex]
```

Comment:
You've created `axisOrder` as a set, unordered but keyed by the `AxisRecord.axisOrdering` field. Yet here you're looking up a tag in `axisOrder` by using `AxisValueFormat1.axisIndex`? Unless I'm missing something, that doesn't seem right. I would think you'd index into `axisRecords` to get the tag.

`axisTag = axisRecords[axisValue.AxisIndex].AxisTag`

Reply:
Hey Peter. I'm still drafting the PR so I'm moving things around quite a bit

Atm, it may be best to just review the unit tests.